### ISSUE TITLE
[Bug]: DuckDuckGo doesn't always display suggestions

### ISSUE THREAD

**Issue by AndiAJ** [mozilla-mobile/fenix#25662]
### Prerequisites
"Show search engines" should be enabled in the "Search" settings

### Steps to reproduce
1. Tap the toolbar
2. Select DuckDuckGo from the "Search engine" shortcut list
3. Set the toolbar text to "mozilla"
4. Check the awesome bar for suggestions

### Expected behaviour
The suggestions should be displayed

### Actual behaviour
Sometimes no suggestions are displayed

### Device name
Google Pixel 3a

### Android version
Android 11

### Firefox release type
Firefox Nightly 103.0a1

### Additional information
❗ In soma cases it happens on first run, in other cases only after you clear the toolbar and re-type the String.
► Video

https://user-images.githubusercontent.com/51314259/174320602-8dc27939-1b6a-4cf4-a705-4909245f9c29.mp4

**Comment by AndiAJ** [comment]
Closing in favour of #23864
